cluster.go

```diff
@@ -1031,6 +1036,66 @@ func (c *Cluster) pin(pin api.Pin, blacklist []peer.ID, prioritylist []peer.ID)
 	if err := isReplicationFactorValid(rplMin, rplMax); err != nil {
 		return false, err
 	}
+	switch pin.Type {
+	case api.DataType:
+		// Fall through
+	case api.ShardType:
+		if !pin.Recursive {
+			return false, errors.New("must pin shards recursively")
+		}
+		if len(pin.Parents) > 1 {
+			return false, errors.New("up to one parent per commitment of a shard")
+		}
+		// In general multiple clusterdags may reference the same shard
+		// and sharder sessions typically update a shard pin's metadata.
+		// Hence we check for an existing shard and carefully update.
+		var empty bool // skip check when state does not yet exist
+		cState, err := c.consensus.State()
+		if err != nil {
+			if strings.Contains(err.Error(), "no state has been agreed upon yet") {
```

Comment:
Can we make "no estate has been agreed upon yet" an `ErrNoState` in consensus and check against it?